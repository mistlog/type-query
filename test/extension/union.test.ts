import { getResultCode } from "../common";

describe("union", () => {
    test("to tuple", () => {
        const name = "union-to-tuple";
        const code = getResultCode(name);
        expect(code).toMatchSnapshot();
    })

    test("to intersection", () => {
        const name = "union-to-intersection";
        const code = getResultCode(name);
        expect(code).toMatchSnapshot();
    })
})