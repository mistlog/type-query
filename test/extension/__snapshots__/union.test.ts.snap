// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`union to intersection 1`] = `
"type input = (\\"foo\\" | 42 | true);
type result = (\\"foo\\" & 42 & true);"
`;

exports[`union to tuple 1`] = `
"type input = (1 | \\"a\\" | true);
type result = [1, \\"a\\", true];"
`;
